bech32m/codecs.py

```diff
@@ -73,21 +74,21 @@ def bech32_verify_checksum(hrp, data):
     raise DecodeError()
 
 
-def bech32_create_checksum(hrp, data, spec):
+def bech32_create_checksum(hrp: str, data: bytes, spec: Encoding) -> bytes:
     """Compute the checksum values given HRP and data."""
     values = bech32_hrp_expand(hrp) + data
     const = BECH32M_CONST if spec == Encoding.BECH32M else 1
-    polymod = bech32_polymod([*values, 0, 0, 0, 0, 0, 0]) ^ const
-    return [(polymod >> 5 * (5 - i)) & 31 for i in range(6)]
+    polymod = bech32_polymod(values + bytes(6)) ^ const
+    return bytes((polymod >> 5 * (5 - i)) & 31 for i in range(6))
 
 
-def bech32_encode(hrp, data, spec):
+def bech32_encode(hrp: str, data: bytes, spec: Encoding) -> str:
     """Compute a Bech32 string given HRP and data values."""
     combined = data + bech32_create_checksum(hrp, data, spec)
     return hrp + "1" + "".join([CHARSET[d] for d in combined])
 
 
-def bech32_decode(bech):
+def bech32_decode(bech: str) -> tuple[str, memoryview, Encoding]:
```

Comment:
_nit._ Consider `NewType`s or `TypeAlias`es for the HRP:

```suggestion
def bech32_decode(bech: str) -> tuple[Hrp, memoryview, Encoding]:
```

Reply:
But it's supplied in the public interface too, so it would be one more breaking change. Moreover, it would be inconvenient to wrap it:
```python
witver, decoded = bech32m.decode(bech32.Hrp(hrp), addr)
```

Reply:
Hence `TypeAlias`? But actually, a `NamedTuple` with `hrp: str` will solve an issue just as good